src/Handler/CurlFactory.php

```diff
@@ -169,7 +169,7 @@ private static function createRejection(EasyHandle $easy, array $ctx): PromiseIn
 
         // If an exception was encountered during the onHeaders event, then
         // return a rejected promise that wraps that exception.
-        if ($easy->onHeadersException) {
+        if ($easy->response && $easy->onHeadersException) {
```

Comment:
I'm not sure if this is the correct fix for the PHPStan error, as there will always be a `response` when `onHeadersException` is set.

We could do something like this instead https://github.com/davidgrayston/guzzle/commit/76d370b96d3e289bf910234fa002ba0d2e2293ea, which makes a few assumptions:
- No response = `ConnectException`
- There's a future possibility that `onHeadersException` can be present without a response (seems unlikely)